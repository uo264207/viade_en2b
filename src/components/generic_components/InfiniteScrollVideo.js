import React from "react";
import InfiniteScroll from "react-infinite-scroll-component";


class MyVideoInfiniteScroll extends React.Component {
  state = {
    items: this.props.content.slice(0,20),
    array_index: 20,
    hasMore: true
  };

  fetchMoreData = () => {
    // a fake async api call like which sends
    // 20 more records in 1.5 secs
    setTimeout(() => {
      this.setState({
        items: this.state.items.concat(this.props.content.slice(this.state.array_index, this.state.array_index+20)),
        array_index: this.state.array_index+20,
      });
    }, 1000);
    if(this.props.content.length<=this.state.array_index){
        this.setState({
            hasMore:false
        });
    }
    console.log(this.state.array_index)
  };

  render() {
    return (
      <div>
        <InfiniteScroll
          dataLength={this.state.items.length}
          next={this.fetchMoreData}
          hasMore={this.state.hasMore}
          loader={<h4>Estoy cargando moreno</h4>}
          endMessage = {
            <p style={{textAlign: 'center'}}>
                <b>Yay! You have seen it all</b>
            </p>
          }
          scrollableTarget= {this.props.scrollParent}

        >
          {this.state.items.map((i, index) => (
            <div key={index}>
              <video width="320" height="240" controls>
                <source src={i.filePath} type="video/mp4"/>
                Your browser does not support the video tag.
              </video> 
                <span>
                  Published: {i.dateAttached.toDateString()}, {i.dateAttached.getHours()}:{i.dateAttached.getMinutes()}
                </span>     
            </div>
          ))}
        </InfiniteScroll>
      </div>
    );
  }
}

export default MyVideoInfiniteScroll;