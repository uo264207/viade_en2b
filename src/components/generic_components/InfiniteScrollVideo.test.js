import React, {Component} from "react";
import InfiniteScrollVideo from "./InfiniteScrollVideo";
import Adapter from "enzyme-adapter-react-16";
import Enzyme, { shallow, mount } from "enzyme";

Enzyme.configure({ adapter: new Adapter() });

test("Definition",()=>{
    expect(InfiniteScrollVideo).toBeDefined();
});

/////////////////////////

test("Render Correct",()=>{
    const scroll = (<InfiniteScrollVideo content={{name:"hola",dateAttached:new Date("January 22, 2012 06:32:32")}}/>);
    expect(scroll).toMatchSnapshot();
});

/////////////////////////

test("Test Empty Prop",()=>{
    const scroll = (<InfiniteScrollVideo content={{name:"hola",dateAttached:new Date("January 22, 2012 06:32:32")}}/>);
    expect(scroll.props.message).toBeUndefined();
});

/////////////////////////

test("Test State",()=>{
    const scroll = mount(<InfiniteScrollVideo content={[{name:"hola",dateAttached:new Date("January 22, 2012 06:32:32")}]}/>);
    scroll.setState({hasMore: false});
    expect(scroll.state().hasMore).toBe(false);
});

/////////////////////////

test("Test Seen all",()=>{
    const scroll = mount(<InfiniteScrollVideo content={[{name:"hola",dateAttached:new Date("January 22, 2012 06:32:32")}]}/>);
    const element = scroll.find("InfiniteScroll");
    expect(element.exists).toBeTruthy();
});

