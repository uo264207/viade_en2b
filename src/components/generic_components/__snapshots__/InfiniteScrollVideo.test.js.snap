// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Correct 1`] = `
<MyVideoInfiniteScroll
  content={
    Object {
      "dateAttached": 2012-01-22T06:32:32.000Z,
      "name": "hola",
    }
  }
/>
`;
